src/vs/workbench/browser/web.main.ts

```diff
@@ -203,8 +203,7 @@ class BrowserMain extends Disposable {
 		]);
 
 		// Workspace Trust Service
-		// TODO @lszomoru: Following two services shall be merged into single service
-		const workspaceTrustManagementService = new WorkspaceTrustManagementService(configurationService, storageService, uriIdentityService, configurationService, environmentService);
+		const workspaceTrustManagementService = new WorkspaceTrustManagementService(configurationService, environmentService, storageService, uriIdentityService, configurationService);
```

Comment:
Ideally we keep service registration in the top level entry files (e.g. [here](https://github.com/microsoft/vscode/blob/73c85159e72069f69db46fad178f37b5d015f5bc/src/vs/workbench/workbench.common.main.ts#L96-L96)) unless one of the core services depend on them.

Reply:
We had to instantiate the `WorkspaceTrustManagementService` here as @sandy081, needs to take a dependency on the `WorkspaceTrustManagementService` which in itself has a dependency on the configuration service. Will let him chime in as well in case I missed anything.

Reply:
Maybe this needs a similar hack like this one then:

https://github.com/microsoft/vscode/blob/e51173f3b6f18f67a2ab9937374aa771f0483665/src/vs/workbench/browser/workbench.ts#L203-L209

Reply:
@sandy081, thoughts?

Reply:
`WorkspaceTrustManagementService` is a core service that `WorkspaceService` (aka configuration service) needs to depend on before it initializes it configuration, because the configuration can change depending on the workspace trust. So this service cannot be acquired later like `InstantiationService`. 

To make it complex, `WorkspaceTrustManagementService` depends on `StorageService` so we cannot instantiate `WorkspaceTrustManagementService` before `StorageService` is created and initialized. `StorageService` and `WorkspaceService` are created and initialized in parallel. Unless we compromise on start up performance (which we do not want to) we cannot pass `WorkspaceTrustManagementService` to `WorkspaceService` in constructor. 

Hence `WorkspaceTrustManagementService` is created after `StorageService` and `WorkspaceService` are created and initialized and workspace trust of `WorkspaceService` is updated from outside as below

https://github.com/microsoft/vscode/blob/6779f85321456851940ea2b79233089d78596f1f/src/vs/workbench/electron-sandbox/shared.desktop.main.ts#L269-L271